docs/src/modules/ROOT/pages/optimization-algorithms/overview.adoc

```diff
@@ -637,6 +637,126 @@ This is useful for benchmarking.
+The first window is called a grace period, which cannot trigger a termination.
+
+image::optimization-algorithms/overview/diminishedReturnsTermination.png[align="center"]
+
+[NOTE]
+====
+If the hard score improves, the grace period is reset and the initial rate of improvement is recalculated.
+====
+
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns />
+    </termination>
+  </localSearch>
+----
+
+There are two properties that can optionally be configured:
+
+* `slidingWindowDuration`, which is the initial grace period and is the length of the window that is used for the rate of improvement calculation.
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowDuration>PT2M30S</slidingWindowDuration>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
+
+Alternatively to a `java.util.Duration` in ISO 8601 format, you can also use:
++
+** Milliseconds
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowMilliseconds>500</slidingWindowMilliseconds>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
++
+** Seconds
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowSeconds>500</slidingWindowSeconds>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
++
+** Minutes
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowMinutes>500</slidingWindowMinutes>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
++
+** Hours
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowHours>500</slidingWindowHours>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
++
+** Days
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <slidingWindowDays>500</slidingWindowDays>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
+
+* `minimumImprovementRatio`, which is the minimum value that the ratio between the current and the initial rate of change can be to prevent termination.
+For instance, if `minimumImprovementRatio` is `0.2`, then the current rate of change must be `20%` of the initial rate of change to prevent termination.
++
+[source,xml,options="nowrap"]
+----
+  <localSearch>
+    <termination>
+      <diminishedReturns>
+        <minimumImprovementRatio>0.2</minimumImprovementRatio>
+      </diminishedReturns>
+    </termination>
+  </localSearch>
+----
+
+This `Termination` can only be used for a `Phase` (such as ``<localSearch>``), not for the `Solver` itself.
```

Comment:
I'd invert this statement - this termination cannot be used globally on the solver, only on a phase, and not on a CH phase.

Reply:
Copied from step count termination https://docs.timefold.ai/timefold-solver/latest/optimization-algorithms/overview#stepCountTermination

Reply:
And it can be used on a CH phase (no terminations are exclusive to LS; all phase exclusive terminations will work on any phase). We are also inconsistent in how the unimproved terminations handle it; unimproved time spent is allowed to be configured on a solver level (and just ignores the CH: https://github.com/TimefoldAI/timefold-solver/blob/4b446075bea81eea393edd887a32960470e2970b/core/src/main/java/ai/timefold/solver/core/impl/solver/termination/UnimprovedTimeMillisSpentScoreDifferenceThresholdTermination.java#L61-L71), whereas unimproved step count must be configured on phases (https://github.com/TimefoldAI/timefold-solver/blob/4b446075bea81eea393edd887a32960470e2970b/core/src/main/java/ai/timefold/solver/core/impl/solver/termination/UnimprovedStepCountTermination.java#L25-L26).

Reply:
This is interesting - the inconsistency is surprising me. I thought the fail-fast was present in all the unimproved terminations. We'll get to fix that eventually.